drgn_tools/task.py

```diff
@@ -185,6 +195,173 @@ def get_command(task: Object) -> str:
+
+        rss_task += task_filerss + task_anonrss + task_shmrss
+
+        gid = task.tgid.value_()
+        try:
+            gid_rss = rss_task_dict_global[gid]
+        except KeyError:
+            gid_rss = 0
+        else:
+            rss_task += gid_rss
+
+        rss_task_dict_global[gid] = rss_task
+
+        try:
+            filerss = task.mm.rss_stat.count[MM_FILEPAGES].counter.value_()
+            anonrss = task.mm.rss_stat.count[MM_ANONPAGES].counter.value_()
+            if MM_SHMEMPAGES > 0:
+                shmrss = task.mm.rss_stat.count[MM_SHMEMPAGES].counter.value_()
+            else:
+                shmrss = 0
+
+            rss_mm = filerss + anonrss + shmrss
+        except drgn.FaultError:
+            rss_mm = 0
+
+        rss_mm_dict_global[task.pid.value_()] = rss_mm
+
+
+def get_rss(prog: Program, task: Object) -> int:
+    """
+    :returns RSS value for a task
+    should be used for single task
+    """
+    MM_FILEPAGES = prog.constant("MM_FILEPAGES").value_()
+    MM_ANONPAGES = prog.constant("MM_ANONPAGES").value_()
+    try:
+        MM_SHMEMPAGES = prog.constant("MM_SHMEMPAGES").value_()
+    except LookupError:
+        MM_SHMEMPAGES = -1
+
+    try:
+        filerss = task.mm.rss_stat.count[MM_FILEPAGES].counter.value_()
+        anonrss = task.mm.rss_stat.count[MM_ANONPAGES].counter.value_()
+        if MM_SHMEMPAGES > 0:
+            shmrss = task.mm.rss_stat.count[MM_SHMEMPAGES].counter.value_()
+        else:
+            shmrss = 0
+
+        rss_mm = filerss + anonrss + shmrss
+
+    except drgn.FaultError:
+        rss_mm = 0
+
+    gid = task.tgid.value_()
+
+    rss_task = rss_task_dict.get(gid)
+
+    if gid in rss_task_dict:
+        rss_task = rss_task_dict[gid]
+    else:
+        rss_task = 0
+        ltask = task.group_leader
+
+        task_filerss = ltask.rss_stat.count[MM_FILEPAGES].value_()
+        task_anonrss = ltask.rss_stat.count[MM_ANONPAGES].value_()
+        if MM_SHMEMPAGES > 0:
+            task_shmrss = ltask.rss_stat.count[MM_SHMEMPAGES].value_()
+        else:
+            task_shmrss = 0
+
+        rss_task += task_filerss + task_anonrss + task_shmrss
+
+        for gtask in list_for_each_entry(
+            "struct task_struct",
+            task.group_leader.thread_group.address_of_(),
+            "thread_group",
+        ):
+            task_filerss = gtask.rss_stat.count[MM_FILEPAGES].value_()
+            task_anonrss = gtask.rss_stat.count[MM_ANONPAGES].value_()
+            if MM_SHMEMPAGES > 0:
+                task_shmrss = gtask.rss_stat.count[MM_SHMEMPAGES].value_()
+            else:
+                task_shmrss = 0
+
+            rss_task += task_filerss + task_anonrss + task_shmrss
+
+        rss_task_dict[gid] = rss_task
+
+    rss = rss_mm + rss_task
+    page_size = prog["PAGE_SIZE"].value_()
+    rss_kb = (rss * page_size) // ByteToKB
+
+    return rss_kb
+
+
+def get_pct_mem(prog: Program, task: Object) -> str:
+    """
+    :returns %MEM value
```

Comment:
If you're going to have a one-line docstring, it should be like the following:

```suggestion
    Return the percent memory usage value
```

The `:returns:` keyword can be used in multiline docstrings to explicitly document a return value (but then it would need both colons).